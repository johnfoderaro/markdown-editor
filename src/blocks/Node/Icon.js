import styled from 'styled-components';

const Icon = styled.div`
  padding: .25rem;
  align-self: center;
`;

export default Icon;
