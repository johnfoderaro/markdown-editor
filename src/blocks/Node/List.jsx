import styled from 'styled-components';

const List = styled.div`
  margin-top: 1rem;
  margin-bottom: 1rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  display: flex;
  flex-direction: column;
`;

export default List;
