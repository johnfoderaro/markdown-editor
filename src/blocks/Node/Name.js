import styled from 'styled-components';

const Name = styled.div`
  padding: 2rem;
`;

export default Name;
