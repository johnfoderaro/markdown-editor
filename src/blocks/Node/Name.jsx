import styled from 'styled-components';

const Name = styled.div`
  padding: .25rem;
`;

export default Name;
