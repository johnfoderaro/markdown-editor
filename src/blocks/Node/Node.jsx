import styled from 'styled-components';

import Name from './Name';
import Icon from './Icon';
import Arrow from './Arrow';
import Dir from './Dir';
import File from './File';
import List from './List';

const Node = styled.div`
  display: flex;
  flex-direction: row;
`;

Node.List = List;
Node.Name = Name;
Node.Icon = Icon;
Node.Arrow = Arrow;
Node.Dir = Dir;
Node.File = File;

export default Node;
