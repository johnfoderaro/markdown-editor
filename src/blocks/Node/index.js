import styled from 'styled-components';

import Name from './Name';
import Icon from './Icon';

const Node = styled.div`
  display: flex;
  flex-direction: row;
`;

Node.Name = Name;
Node.Icon = Icon;

export default Node;
