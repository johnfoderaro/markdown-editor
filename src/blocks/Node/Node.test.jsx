import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import Node from './Node';

let tree;

beforeAll(() => {
  tree = mount(
    <Node>
      <Node.Name>
        <h1>Node Icons</h1>
      </Node.Name>
      <Node.Icon>
        <Node.Arrow width="20" height="20" />
      </Node.Icon>
      <Node.Icon>
        <Node.Dir width="20" height="20" state="open" />
      </Node.Icon>
      <Node.Icon>
        <Node.Dir width="20" height="20" state="closed" />
      </Node.Icon>
      <Node.Icon>
        <Node.File width="20" height="20" />
      </Node.Icon>
    </Node>,
  );
});

describe('<Node />', () => {
  it('should be defined', () => {
    expect(tree).toBeDefined();
  });
  it('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });
  it('should contain a <Node.Name /> element', () => {
    expect(tree.find(Node.Name)).toBeDefined();
  });
  it('should contain a <Node.Icon /> element', () => {
    expect(tree.find(Node.Icon)).toBeDefined();
  });
  it('should render an SVG icons', () => {
    expect(tree.find(Node.Arrow)).toBeDefined();
    expect(tree.find(Node.File)).toBeDefined();
    expect(tree.find(Node.Dir)).toBeDefined();
  });
});
