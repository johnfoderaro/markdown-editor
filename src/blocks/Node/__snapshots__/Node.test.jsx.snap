// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Node /> should render correctly 1`] = `
.c1 {
  padding: .25rem;
}

.c2 {
  padding: .25rem;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<styled.div>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "Arrow": [Function],
        "Dir": [Function],
        "File": [Function],
        "Icon": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "c2",
            "rules": Array [
              "
  padding: .25rem;
  align-self: center;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "List": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "rules": Array [
              "
  margin-top: 1rem;
  margin-bottom: 1rem;
  padding-top: 1rem;
  padding-bottom: 1rem;
  display: flex;
  flex-direction: column;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "Name": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c1",
            "rules": Array [
              "
  padding: .25rem;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "lastClassName": "c0",
          "rules": Array [
            "
  display: flex;
  flex-direction: row;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    >
      <styled.div>
        <StyledComponent
          forwardedClass={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "lastClassName": "c1",
                "rules": Array [
                  "
  padding: .25rem;
",
                ],
              },
              "displayName": "styled.div",
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c1"
          >
            <h1>
              Node Icons
            </h1>
          </div>
        </StyledComponent>
      </styled.div>
      <styled.div>
        <StyledComponent
          forwardedClass={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "lastClassName": "c2",
                "rules": Array [
                  "
  padding: .25rem;
  align-self: center;
",
                ],
              },
              "displayName": "styled.div",
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "div",
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c2"
          >
            <Arrow
              height="20"
              width="20"
            >
              <svg
                height="20"
                viewBox="0 0 1792 1792"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1675 971q0 51-37 90l-75 75q-38 38-91 38-54 0-90-38l-294-293v704q0 52-37.5 84.5t-90.5 32.5h-128q-53 0-90.5-32.5t-37.5-84.5v-704l-294 293q-36 38-90 38t-90-38l-75-75q-38-38-38-90 0-53 38-91l651-651q35-37 90-37 54 0 91 37l651 651q37 39 37 91z"
                />
              </svg>
            </Arrow>
          </div>
        </StyledComponent>
      </styled.div>
      <styled.div>
        <StyledComponent
          forwardedClass={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "lastClassName": "c2",
                "rules": Array [
                  "
  padding: .25rem;
  align-self: center;
",
                ],
              },
              "displayName": "styled.div",
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "div",
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c2"
          >
            <Dir
              height="20"
              state="open"
              width="20"
            >
              <svg
                height="20"
                viewBox="0 0 2048 1792"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1845 931q0-35-53-35h-1088q-40 0-85.5 21.5t-71.5 52.5l-294 363q-18 24-18 40 0 35 53 35h1088q40 0 86-22t71-53l294-363q18-22 18-39zm-1141-163h768v-160q0-40-28-68t-68-28h-576q-40 0-68-28t-28-68v-64q0-40-28-68t-68-28h-320q-40 0-68 28t-28 68v853l256-315q44-53 116-87.5t140-34.5zm1269 163q0 62-46 120l-295 363q-43 53-116 87.5t-140 34.5h-1088q-92 0-158-66t-66-158v-960q0-92 66-158t158-66h320q92 0 158 66t66 158v32h544q92 0 158 66t66 158v160h192q54 0 99 24.5t67 70.5q15 32 15 68z"
                />
              </svg>
            </Dir>
          </div>
        </StyledComponent>
      </styled.div>
      <styled.div>
        <StyledComponent
          forwardedClass={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "lastClassName": "c2",
                "rules": Array [
                  "
  padding: .25rem;
  align-self: center;
",
                ],
              },
              "displayName": "styled.div",
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "div",
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c2"
          >
            <Dir
              height="20"
              state="closed"
              width="20"
            >
              <svg
                height="20"
                viewBox="0 0 1792 1792"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1600 1312v-704q0-40-28-68t-68-28h-704q-40 0-68-28t-28-68v-64q0-40-28-68t-68-28h-320q-40 0-68 28t-28 68v960q0 40 28 68t68 28h1216q40 0 68-28t28-68zm128-704v704q0 92-66 158t-158 66h-1216q-92 0-158-66t-66-158v-960q0-92 66-158t158-66h320q92 0 158 66t66 158v32h672q92 0 158 66t66 158z"
                />
              </svg>
            </Dir>
          </div>
        </StyledComponent>
      </styled.div>
      <styled.div>
        <StyledComponent
          forwardedClass={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "lastClassName": "c2",
                "rules": Array [
                  "
  padding: .25rem;
  align-self: center;
",
                ],
              },
              "displayName": "styled.div",
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "div",
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="c2"
          >
            <FileIcon
              height="20"
              width="20"
            >
              <svg
                height="20"
                viewBox="0 0 1792 1792"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1596 380q28 28 48 76t20 88v1152q0 40-28 68t-68 28h-1344q-40 0-68-28t-28-68v-1600q0-40 28-68t68-28h896q40 0 88 20t76 48zm-444-244v376h376q-10-29-22-41l-313-313q-12-12-41-22zm384 1528v-1024h-416q-40 0-68-28t-28-68v-416h-768v1536h1280zm-1024-864q0-14 9-23t23-9h704q14 0 23 9t9 23v64q0 14-9 23t-23 9h-704q-14 0-23-9t-9-23v-64zm736 224q14 0 23 9t9 23v64q0 14-9 23t-23 9h-704q-14 0-23-9t-9-23v-64q0-14 9-23t23-9h704zm0 256q14 0 23 9t9 23v64q0 14-9 23t-23 9h-704q-14 0-23-9t-9-23v-64q0-14 9-23t23-9h704z"
                />
              </svg>
            </FileIcon>
          </div>
        </StyledComponent>
      </styled.div>
    </div>
  </StyledComponent>
</styled.div>
`;
